### PR TITLE
Add purge private data documentation

### PR DESCRIPTION
**Pull request by denyeart**
Add docs for the v2.5 purge private data functionality.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

### CODE REVIEW

File: docs/source/private-data-arch.rst [hunk 1]
```diff
@@ -380,10 +380,28 @@ Considerations when using private data
...
+**Purging private data in chaincode**
+
+Private data can be deleted from state just like regular state data so that it is not available for query in chaincode for future transactions.
+However, when private data is simply deleted from state, the history of the private data remains in the peer's private database so that it can be returned in block events and returned to other peers that are catching up to the current block height.
+If you need to completely remove the private data from all peers that have access to it, use the chaincode API ``PurgePrivateData`` instead of the ``DelPrivateData`` API.
+
+The ``PurgePrivateData`` chaincode API is available starting in Fabric v2.5. To ensure that all peers are at v2.5 or later, the application capability ``V2_5`` or higher must be set in the channel's configuration before using the feature.
+
+Private data is purged from a peer's private database at the time of block commit.
+For more efficiency, the purge requests can be processed at certain block intervals based on the ``ledger.pvtdataStore.purgeInterval`` setting in the peer ``core.yaml`` configuration.
```
Comment (jt-nti) [inline docs/source/private-data-arch.rst]: I think there's something missing here clarifying what happens at commit time and what background processing occurs later, and how to confirm when it has completed successfully.

File: docs/source/private-data-arch.rst [hunk 2]
```diff
@@ -380,10 +380,28 @@ Considerations when using private data
...
+To process purge requests every block set ``ledger.pvtdataStore.purgeInterval`` to ``1``.
```
Comment (jt-nti) [inline docs/source/private-data-arch.rst:395]: What's the default?

**Review by jt-nti — COMMENTED** [review]
The purge in chaincode vs automatic purging is clear which is good. I think more detail about the background processing of purge requests would be useful.

File: docs/source/private-data-arch.rst (hunk 1 above)
Comment (denyeart) [reply docs/source/private-data-arch.rst]: Clarified the commit versus purgeInterval processing.

File: docs/source/private-data-arch.rst (hunk 2 above)
Comment (denyeart) [reply docs/source/private-data-arch.rst:395]: Clarified default is 100.

**Comment by denyeart** [comment]
Thanks @jt-nti , I've added clarifications in latest push.

**Review by ale-linux — APPROVED** [review]

**Comment by denyeart** [comment]
@Mergifyio backport release-2.5

**Pull request merged**